sentinel-adapter/sentinel-apache-dubbo-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/dubbo/SentinelDubboProviderFilter.java

```diff
@@ -60,16 +70,14 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
         Entry interfaceEntry = null;
         Entry methodEntry = null;
         try {
-            String methodResourceName = DubboUtils.getResourceName(invoker, invocation, DubboConfig.getDubboProviderPrefix());
-            String interfaceResourceName = DubboConfig.getDubboInterfaceGroupAndVersionEnabled() ? invoker.getUrl().getColonSeparatedKey()
-                    : invoker.getInterface().getName();
+            String methodResourceName = getMethodName(invoker, invocation);
+            String interfaceResourceName = getInterfaceName(invoker);
             // Only need to create entrance context at provider side, as context will take effect
             // at entrance of invocation chain only (for inbound traffic).
             ContextUtil.enter(methodResourceName, application);
             interfaceEntry = SphU.entry(interfaceResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.IN);
-            rpcContext.set(DubboUtils.DUBBO_INTERFACE_ENTRY_KEY, interfaceEntry);
             methodEntry = SphU.entry(methodResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.IN, invocation.getArguments());
-            rpcContext.set(DubboUtils.DUBBO_METHOD_ENTRY_KEY, methodEntry);
+            rpcContext.set(methodResourceName, new Entry[]{interfaceEntry, methodEntry});
```

Comment:
This may cause leak of `interfaceEntry` when the `methodEntry` is blocked. Refer https://github.com/alibaba/Sentinel/pull/1124#discussion_r354810059

Reply:
+1, when `methodEntry` is blocked, the `rpcContext#set` cann't be executed, and the value in `rpcContext` with key of `methodResourceName` is null. We may need to keep the same as the previous commit, separate with two different key.